memlimit/memlimit.go

```diff
@@ -29,56 +30,106 @@ var (
 	logger = log.New(io.Discard, "", log.LstdFlags)
 )
 
+type config struct {
+	printf   func(string, ...interface{})
+	ratio    float64
+	provider Provider
+}
+
+// An Option alters the behavior of SetGoMemLimitWithEnv.
+type Option interface {
+	apply(*config)
+}
+
+type optionFunc func(*config)
+
+func (of optionFunc) apply(cfg *config) { of(cfg) }
```

Comment:
What about simplifying it to `type Option func(cfg *config)`?

Reply:
@KimMachineGun, totally agree